src/main/java/bio/terra/service/filedata/google/firestore/FireStoreDirectoryDao.java

```diff
@@ -627,16 +606,18 @@ private DocumentSnapshot lookupByPathNoXn(
                 .document(encodePathAsFirestoreDocumentName(lookupPath));
 
         RuntimeException lastException = null;
-        for (int retryNum = 0; retryNum < LOOKUP_RETRIES; retryNum++) {
+        for (int retryNum = 0; retryNum < fireStoreUtils.getFirestoreRetries(); retryNum++) {
             logger.info("FirestoreDirectoryDao lookupByPathNoXn - iteration {}", retryNum);
             try {
                 ApiFuture<DocumentSnapshot> docSnapFuture = docRef.get();
                 return docSnapFuture.get();
             } catch (AbortedException | ExecutionException ex) {
                 lastException = fireStoreUtils.handleExecutionException(ex, "lookupByPathNoXn");
             }
-            TimeUnit.SECONDS.sleep(LOOKUP_WAIT_SECONDS);
+            final long retryWait = SLEEP_BASE_MILLISECONDS * Double.valueOf(Math.pow(2.5, retryNum)).longValue();
+            TimeUnit.SECONDS.sleep(retryWait);
         }
+        logger.error("Ran out of retries - lookupPathNoXn");
         throw lastException;
```

Comment:
Note that this will NPE if someone sets `FireStoreutils.firestoreRetries` to zero. You may want to add a check in the `FireStoreUtils` constructor and force it to `1` if it's less than `1`.

Reply:
Thanks - I'm wondering where the best place is to put this check. Would GoogleResourceConfiguration.java be better than in FirestoreUtils? Might as well get it correctly from the source?

Reply:
It looks like there's not much logic in our configuration classes. I can see the check going into either place. I think I like `FirestoreUtils` better because that will put the check closer to the reason for the check.

Looking at the code in `FirestoreUtils`, it looks like it's written so that at least one attempt will occur even if `firestoreRetries` is `0`. So maybe this code should be changed to work that way too, so no minimum value is needed.

Reply:
Ok, I've changed the FirestoreDirectoryDao so that it always runs at least once, even if the max is zero.